Assets/Scripts/Fire.cs

```diff
@@ -0,0 +1,38 @@
+﻿using UnityEngine;
+
+public class Fire : MonoBehaviour {
+    public float scaleFire;
+    public float speedDreasing;
+    public float minScale;
+    public int densityFire;
+    private ParticleSystem part;
+    private ParticleSystem.ShapeModule sh;
+    private ParticleSystem.EmissionModule em;
+
+    void Start() {
+        part = this.GetComponent<ParticleSystem>();
+        sh = part.shape;
+        em = part.emission;
+        sh.scale = Vector3.ClampMagnitude(Vector3.one, scaleFire);
+        em.rateOverTime = (ParticleSystem.MinMaxCurve)(System.Math.Pow(scaleFire, 3) * densityFire);
+    }
+
+    void Update() {
+        Collider[] woods = Physics.OverlapSphere(this.transform.position, sh.scale.x + 1);
+        foreach (var wood in woods) {
+            if (!part.isStopped && wood.gameObject.GetComponent<Wood>()) {
+                sh.scale += Vector3.ClampMagnitude(Vector3.one, wood.gameObject.GetComponent<Wood>().size);
+                em.rateOverTime = (ParticleSystem.MinMaxCurve)(System.Math.Pow(sh.scale.magnitude, 3));
+                Destroy(wood.gameObject);
+            }
+        }
+
+        if (!part.isStopped && sh.scale.magnitude < minScale) {
+            part.Stop();
+        } else if (!part.isStopped) {
+            sh.scale -= Vector3.ClampMagnitude(Vector3.one, speedDreasing * Time.deltaTime);
+            em.rateOverTime = (ParticleSystem.MinMaxCurve)(System.Math.Pow(sh.scale.magnitude, 3) * densityFire);
+        }
+
+    }
```

Comment:
Small thing but it would be nice if you could add a gizmo to the fire so we can see in the editor how big it is (see end of Crafter as an example)

Reply:
Maybe tomorrow if I have time